### ISSUE TITLE
Testing efficiency: Silver requirement?

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/silver#39]
One of the things I didn't like about WCAG 2.1 is that it has increased the cost of doing accessibility. I don't believe improved readability and a better learning curve should be the only answers that Silver has to improving adoption of accessibility guidelines. From the sound of it, Silver is going to be vastly more complicated than WCAG 2 is, even if we figure out a way to make that more readable, it'll mean far more work, more costs, and more resistance to adopt.

If we want greater adoption, we need to cut out the parts that cost a lot of time. That means trade-offs, fewer exceptions, and a focus on tests that can be scaled. Just to give one example: Someone can very quickly test that links to the same page have the same link text, and that links to different pages don't have the same text. It's much slower to test if all links have a descriptive name within their context. 

There are a lot of requirements in WCAG 2 that are difficult to test, because we needed them to take all sorts of exceptions into account. Very little concern was ever given to how efficient something could be tested. I think for Silver a better balance has to be found between covering all possible edge cases, and making testing efficient and easy to understand.

**Comment by detlevhfischer** [comment]
> a focus on tests that can be scaled. Just to give one example: Someone can very quickly test that links to the same page have the same link text, and that links to different pages don't have the same text. 

My suspicion is that many of the things that scale well are minor compared to show stopper issues which tend to require (part) manual testing. There can be good reasons to have different link test link to the same page, and it is hardly a serious issue, If we want to focus, I think the focus has to be on tests that uncover the most common severe failures. In my experience, the perennial top culprits are 2.1.1 Keyboard (including swipability)  2.4.3 Focus order, 1.1.1 Missing or non-descriptive accNames, 4.1.2 Missing or incorrect role and state information. A focus on low hanging fruit will fail end users.

**Comment by slauriat** [comment]
Apologies for it taking so long to respond! We've had a lot of discussion around this point and others, but have just now started arriving at a point in the content migration efforts that we have real examples to talk and work through as a part of the migration, now that we have a list of SC groupings to build up, and have started expressing them in functional user needs.

In particular, we'd like to work through with you (and anyone else in or out of ACT who can join in TPAC and beyond) how best to build up tests to assess meeting these user needs in a particular technology, with a few aspects of efficiency in mind:
- Time per test
- Effort per test
- Expertise per test
- Automation
- Number of tests per guideline (came up today)

…probably many other aspects as well. I figure we can most effectively move forward in answering this by building up a set of all possible tests (or at least types of tests) for a given set of SC grouping + user needs, now matter how demanding or trivial the test, and then look at the practicality both from a "will anyone do this?" perspective as well as efficiency at scale.

Possible to meet up at TPAC to work through the start of this?

**Comment by jspellman** [comment]
Thank you for your comment. We have closed this issue because we believe this issue has been addressed in FPWD structurally. We hope to work with ACT on future guidelines to improve testing.  We have designed the writing process so writing the tests occurs before writing the guideline.  If you believe this issue should be reopened, please open a new issue and reference this issue. If you have trouble, contact a WCAG 3 editor.
